Sources/Core/Services/NowPlayingService.swift

```diff
@@ -39,31 +39,32 @@ protocol NowPlaying {
 
 final class ModernAVPlayerNowPlayingService: NowPlaying {
 
-    private var infos = [String: Any]()
+    private var infos = [String: Any]() {
+        didSet { MPNowPlayingInfoCenter.default().nowPlayingInfo = infos }
+    }
     private var session: URLSession {
         return URLSession.shared
     }
     private var task: URLSessionTask?
 
     func update(metadata: PlayerMediaMetadata?, duration: Double?, isLive: Bool) {
-        infos = parse(metadata: metadata)
+        var infos = parse(metadata: metadata)
         if #available(iOS 10, *) {
             infos[MPNowPlayingInfoPropertyIsLiveStream] = isLive
         }
         if let duration = duration,
             duration.isNormal {
             infos[MPMediaItemPropertyPlaybackDuration] = duration
         }
-        MPNowPlayingInfoCenter.default().nowPlayingInfo = infos
+        self.infos = infos
     }
 
     func update(metadata: PlayerMediaMetadata) {
-        MPNowPlayingInfoCenter.default().nowPlayingInfo = parse(metadata: metadata)
+        infos = parse(metadata: metadata)
```

Comment:
MPNowPlayingInfoPropertyIsLiveStream & MPMediaItemPropertyPlaybackDuration will be erased so not a good solution. parse should update dictionnary, not create a new one.